src/main/java/com/bandwidth/sdk/messaging/MessagingClient.java

```diff
@@ -76,4 +87,109 @@ public Message sendMessage(SendMessageRequest request) throws ExecutionException
+    }
+
+    /**
+     * Uploads MMS media content. Max 65MB
+     *
+     * @param byteArray byte array of the data to be sent
+     * @param fileName File name as you would like it to be uploaded
+     * @return CompletableFuture that contains URL that can be sent in an MMS
+     */
+    public CompletableFuture<String> uploadMediaAsync(byte[] byteArray, String fileName){
+        String url = MessageFormat.format("{0}/users/{1}/media", MEDIA_URL, userId);
+        try{
+            return httpClient.preparePut(url)
+                    .setBody(byteArray)
+                    .execute()
+                    .toCompletableFuture()
+                    .thenApply((resp) -> {
+                        if (resp.getStatusCode() != 200)
+                            throw new MessagingException(MessageErrorType.fromStatusCode(resp.getStatusCode()).toString());
+                        return MessageFormat.format("{0}/users/{1}/media/{2}", MEDIA_URL, userId, fileName);
+                    });
+        } catch (MessagingException e) {
+            CompletableFuture<String> future = new CompletableFuture<>();
+            future.completeExceptionally(e);
+            return future;
+        }
+    }
+
+    /**
+     * Downloads MMS media content asynchronously .
+     *
+     * @param mediaUrl URL that you would like to download
+     * @return CompletableFuture that contains byte array of the mms content
+     */
+    public CompletableFuture<byte[]> downloadMessageMediaAsync(String mediaUrl){
+        try{
+            return httpClient.prepareGet(mediaUrl)
+                    .execute()
+                    .toCompletableFuture()
+                    .thenApply((resp) -> {
+                        if (resp.getStatusCode() != 200)
+                            throw new MessagingException(MessageErrorType.fromStatusCode(resp.getStatusCode()).toString());
+                        return resp.getResponseBodyAsBytes();
+                    });
+        } catch (MessagingException e) {
+            CompletableFuture<byte[]> future = new CompletableFuture<>();
+            future.completeExceptionally(e);
+            return future;
+        }
+    }
+
+    /**
+     * Downloads MMS media content.
+     *
+     * @param mediaUrl media urls to download from
+     * @return byte array containing the mms content
+     */
+    public byte[] downloadMessageMediaAsBytes(String mediaUrl){
+        return downloadMessageMediaAsync(mediaUrl).join();
+    }
+
+    /**
+     * Downloads MMS media content and stores it to disk
+     *
+     * @param mediaUrl media urls to download from
+     * @param path path on disk to store file to
+     * @return byte array containing the mms content
+     */
+    public void downloadMessageMediaToDisk(String mediaUrl, String path) throws IOException {
+        FileOutputStream stream = new FileOutputStream(path);
+        stream.write(downloadMessageMediaAsBytes(mediaUrl));
```

Comment:
You need to close the stream. You can wrap it in java's "try-with-resources" to make sure it gets closed properly